### ISSUE TITLE
Performance waste due to decrypting each secret 34 times instead of once

### ISSUE THREAD

**Issue by hohwille** [ulisesbocchio/jasypt-spring-boot#7]
I set a breakpoint into EncryptablePropertySource and figured out that it is called many times for the same property. So I added this line inside the inner if block (Line 17):
```
System.out.println("Decrypting value of property " + name);
```

When I startup my simple spring boot application with only a single datasource I get this output:
> Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password
Decrypting value of property spring.datasource.password

This means the same (and single) encrypted secret is decrypted 34 times.

IMHO this is rather a design flaw of spring that your code but I want to point this out and will also connect with the spring boot team on github.

**Comment by ulisesbocchio** [comment]
Yeah, I'm aware of this. The encryptable property source just wraps the original one. If you access the property source N times the property will be decrypted N times (i.e. using the Environment). The rationale behind this decision is that I don't know what kind of custom PropertySource implementations are out there, nor Spring knows, so I can't cache the returned values as a general solution. I could add a feature to enable encrypted properties caching for the cases when you're absolutely sure the underlying PropertySource (i.e. vanilla spring property sources) is not reloadable in any way.
What I'm not sure is why Spring is getting the property 34 times in your app, how are you injecting the property to your data source?

**Comment by hohwille** [comment]
> how are you injecting the property to your data source?

I am just using spring-boot with hibernate. There is no bean in MY code that gets a data-source injected.

**Comment by hohwille** [comment]
For the record: This is fixed in spring-boot and the fix will come with version 1.3.0.
